python/publish/publisher.py

```diff
@@ -305,6 +307,15 @@ def publish_json(self, data: PublishData):
         # provide a reduced version to Github actions
         self._gha.set_output('json', json.dumps(data.reduced(), ensure_ascii=False))
 
+    def publish_summary(self, data: PublishData):
+        if self._settings.job_summary and  os.getenv("GITHUB_STEP_SUMMARY"):
+            try:
+                with open(os.environ["GITHUB_STEP_SUMMARY"], "a+") as summary:
+                    summary.write(f"\n# {data.title}\n\n")
+                    summary.write(f"{data.summary}\n\n")
+            except:
+                self._gha.error(f"Failed to write summary file $GITHUB_STEP_SUMMARY={os.getenv('GITHUB_STEP_SUMMARY')}")
```

Comment:
This code base prefers `'` for strings over `"`, unless in cases where the latter adds some benefits, like this line.

Reply:
Yeah, black prefers double quotes as rule of thumb, I have to diable it in my IDE for this project 😝

Reply:
Looks like black does not like my personal taste, to bad ;-) I might run black over it one day, but not today.